scripts/fn_paradrop.sqf

```diff
@@ -112,88 +118,97 @@ private _dropRunOrigin = getPosATL _vehicle; // calling position from leader wil
 // Between current position and drop target, find angle
 private _dropTarget = [_loc,true] call CBA_fnc_randPosArea;
 
-private _theta = _dropRunOrigin getDir _dropTarget;
-
-// For a known drop point and known radius with azimuth _theta,
-// and radius r given by _dropLength/2,
-// then coordinates in the xy-plane relative to _dropTarget are:
-// +/- r*sin _theta for x (Easting) and +/- r*cos _theta for y (Northing)
-private _xDiff = _dropLength/2 * sin _theta;
-private _yDiff = _dropLength/2 * cos _theta;
-// set all coordinates based on DZ points in xy-plane; z coord does not matter
-// double lengths on second DZ coordinate to ensure straight line flight
-private _dropZoneLocs = [[(_dropTarget select 0) + 2*_xDiff,(_dropTarget select 1) + 2*_yDiff,_dropHeight],
-    [(_dropTarget select 1) - _xDiff,(_dropTarget select 1) - _yDiff,_dropHeight]];
-
-//expand waypoint 
-private _dp = _pilotGroup addWaypoint [_dropZoneLocs select 1,-1,currentWaypoint _pilotGroup + 1,""];
+if (_dropTarget isEqualTo []) exitWith {};
+
+private _fnc_dzLocs = {
+    params ["_dropTarget","_dropRunOrigin","_dropLength"];
+    private _theta = _dropRunOrigin getDir _dropTarget;
+
+    // For a known drop point and known radius with azimuth _theta,
+    // and radius r given by _dropLength/2,
+    // then coordinates in the xy-plane relative to _dropTarget are:
+    // +/- r*cos _theta for x (Northing) and +/- r*sin _theta for x (Easting)
+    // however, closest one is determined dynamically due to quadrant dependence
+
+    private _yDiff = (sin _theta) * _dropLength/2;
+    private _xDiff = (cos _theta) * _dropLength/2;
+    // set all coordinates based on DZ points in xy-plane; z coord does not matter
+    // double lengths on second DZ coordinate to ensure straight line flight
+    [[(_dropTarget select 0) + _xDiff,(_dropTarget select 1) + _yDiff,_dropHeight],
+        [(_dropTarget select 0) - _xDiff,(_dropTarget select 1) - _yDiff,_dropHeight]];
+};
+
+private _dropZoneLocs = [_dropTarget, _dropRunOrigin, _dropLength] call _fnc_dzLocs;
+private _distanceToDZ = [_dropRunOrigin distance (_dropZoneLocs select 0), _dropRunOrigin distance (_dropZoneLocs select 1)];
+private _minDistance = selectMin _distanceToDZ;
+private _indexOfMin = _distanceToDZ find _minDistance;
```

Comment:
If we had left out handling of `_minDistance < _dropLength` then the calculations of lines 142-144 could actually be replaced by a `switch` statement thus:
`_indexOfMin = switch (floor (_azimuth / 90)) do {
    case 0;
    case 1: {1;};
    case 2;
    case 3: {0;};
};`

Which would produce a mild optimization.

Reply:
Actually, forget this entirely.  I tried running this and it went extraordinarily poorly, and I finally diagnosed why.  The issue was never with my math, honestly, because I think I did it right the first time.  

The issue is that `getDir`'s azimuth finding behavior sucks.  The nominal syntax is `pos1 getDir pos2` where the azimuth is measured from `pos1` to `pos2`.  However, upon further testing, the function will unpredictably measure the azimuth from `pos2` to `pos1` instead regardless of function call.  For example, one of my groups `ghost_hawk_1` was approaching the DZ from the southeast quadrant  which should make its nominal azimuth to the target between 90 and 180 degrees.  Debug logs, however, show its measured azimuth at 309.927 - the RECIPROCAL course!   Hence the sine and cosine of the angle come out with the correct magnitudes but incorrect signs.  This effect appears to be completely unpredictable, thus we are left with the distance calculation intact.   

Here are some further test logs:

- xian_1: Approaches from NE quadrant, azimuth should be between 180 and 270, measured azimuth 237.775, valid.
- blackfish_1: Approach from NW quadrant, azimuth should be between 270 and 360, measured azimuth 137.985, implying approach from SE quadrant - reciprocal.
- chinook_1: Approach from SW quadrant, azimuth should be between 0 and 90, measured azimuth 53.4929, valid.

Reply:
So basically I'm closing this note.  Way to go, Bohemia.

Reply:
Upon further reflection, I've figured it out.  Bohemia's getDir uses compass direction and not trigonometric angle.  Compass direction has 0 as North and advances counterclockwise.  I can put together a switch statement as a parametric function to convert it and a separate one to get the correct minimum angle without calculations.